lib/rubygems/resolver/api_specification.rb

```diff
@@ -6,6 +6,17 @@
 # is the name, version, and dependencies.
 
 class Gem::Resolver::APISpecification < Gem::Resolver::Specification
+  ##
+  # We assume that all instances of this class are immutable;
+  # so avoid duplicated generation for performance.
+  @@cache = {}
```

Comment:
Class variables are pretty inefficient, I think on all Ruby implementations.
Should we use a constant or `@ivar` on the class instead here?